idl/arm_forecast.pro

```diff
@@ -109,15 +110,18 @@ names  = reform( summary( 0, * ) )
     	for i = 0, n_elements( names ) - 1 do begin
 			index = where( names( i ) eq noaa_name )
 			index = index( 0 )
+
 			if ( index eq -1 ) then continue ;index = n_elements( noaa_name ) ; i.e., set to an empty string
 			
-    		printf, lun, names( i ) + ' ' + strtrim(mci( i ),2) + ' ' + strtrim(cprob( i ),2) + '(' + c_prob_noaa( index ) + ')' + ' ' + $
-				strtrim(mprob( i ),2) + '(' + m_prob_noaa( index ) + ')' + ' ' + strtrim(xprob( i ),2) + '(' + x_prob_noaa( index ) + ')'
+    		printf, lun, names( i ) + ' ' + strtrim(mci( i ),2) + ' ' + strtrim(cprob_evol( i ),2)  + '(' + strtrim(cprob( i ),2) + ')' + '(' + c_prob_noaa( index ) + ')' + ' ' + $
```

Comment:
if `names` is not a function, or any of the other variables used here, then it should use `[ i ]` instead of `( i )`. In fact, to keep a better and more sustainable IDL code, I would argue all SM should have: `Compile_Opt idl2` in all their functions. This will force all the integers to be long and all the arrays to use `[ ]`.